framework/py/flwr/cli/run/run.py

```diff
@@ -104,14 +108,12 @@ def run(
             redirect_output(captured_output)
 
         # Determine if app is remote
-        app_id = None
+        app_spec = None
         if (app_str := str(app)).startswith("@"):
-            if not re.match(APP_ID_PATTERN, app_str):
-                raise typer.BadParameter(
-                    "Invalid remote app ID. Expected format: '@account_name/app_name'."
-                )
-            app_id = app_str
-        is_remote_app = app_id is not None
+            # Validate app version and ID format
+            parse_app_spec(app_str)
```

Comment:
Maybe it's better to use `_` for the return value, explicitly singalling that "Yes, this function returns a value, but we are intentionally ignoring it."

```suggestion
            _ = parse_app_spec(app_str)
```